core/src/services/hdfs/writer.rs

```diff
@@ -75,16 +76,19 @@ impl oio::Write for HdfsWriter<hdrs::AsyncFile> {
                 return Poll::Ready(res);
             }
 
-            let _ = Pin::new(&mut self.f)
-                .poll_close(cx)
-                .map_err(new_std_io_error);
-
+            let mut f = self.f.take().expect("HdfsWriter must be initialized");
             // Clone client to allow move into the future.
             let tmp_path = self.tmp_path.clone();
             let client = self.client.clone();
             let target_path = self.target_path.clone();
+            // Clone the necessary parts of the context
+            let waker = cx.waker().clone();
 
             self.fut = Some(Box::pin(async move {
+                // Now use the cloned waker in the async block
+                let mut pinned = std::pin::pin!(f);
```

Comment:
We don't need to do this. We can call `f.close().await` directly here.

Reply:
I tried but it shows error 
```
no close method found for AsyncFile in the current scope.
```

Reply:
`import futures::AsyncWriteExt` brought it in scope

Reply:
Thank you!! @Xuanwo 
This solved the problem. 
Can you please explain why this was an issue? 
I am still new to Rust and this would help me improve my understanding of async rust.

Reply:
> Can you please explain why this was an issue? I am still new to Rust and this would help me improve my understanding of async rust.

The `poll_xxx` API is a low-level method for polling a `Future` within the runtime; however, we seldom use async APIs in this manner.

Most of the time, we will use API likse `close().await` in an `async` context. It will be expanded into a generator that internally calls `poll_xxx` until it returns `Ready`.

You can read more about `Future` here: https://doc.rust-lang.org/beta/std/future/trait.Future.html